src/welcome/ProgressiveProfiling.jsx

```diff
@@ -20,27 +20,27 @@ import {
   Form,
   StatefulButton,
   Hyperlink,
-  Spinner,
 } from '@edx/paragon';
 import { Error } from '@edx/paragon/icons';
 
-import { getFieldData, saveUserProfile } from './data/actions';
+import { saveUserProfile } from './data/actions';
 import { welcomePageSelector } from './data/selectors';
 import messages from './messages';
 
 import { RedirectLogistration } from '../common-components';
 import {
-  DEFAULT_REDIRECT_URL, DEFAULT_STATE, FAILURE_STATE, COMPLETE_STATE,
+  DEFAULT_REDIRECT_URL, DEFAULT_STATE, FAILURE_STATE,
 } from '../data/constants';
 import FormFieldRenderer from '../field-renderer';
 import WelcomePageModal from './WelcomePageModal';
 import BaseComponent from '../base-component';
 
 const ProgressiveProfiling = (props) => {
   const {
-    extendedProfile, fieldDescriptions, formRenderState, intl, submitState, showError,
+    formRenderState, intl, submitState, showError,
   } = props;
-
+  const fieldDescriptions = props.location.state.fieldDescriptions.fields;
+  const extendedProfile = props.location.state.fieldDescriptions.extended_profile;
```

Comment:
`const fieldDescriptions = props.location.state.fieldDescriptions.fields;`
`const extendedProfile = props.location.state.fieldDescriptions.extended_profile;`
We can move this to `mapStateToProps` by using `fieldDescriptionSelector` like

`const mapStateToProps = state => ({
 fieldDescriptions: fieldDescriptionSelector(state),
 )};`
 and then get from props.